ext/background/api.js

```diff
@@ -69,12 +102,29 @@ async function getSessions() {
   return sessions;
 }
 
-// Returns a list of authenticated org slugs from Range cookies
-function orgsFromCookies() {
+// Returns a map of Range API cookie slugs and cookie values. Filters cookies
+// that have been found to be invalid or expired
+function cookieMap() {
   return new Promise((resolve, reject) => {
     chrome.cookies.getAll({ domain: CONFIG.cookie_host || CONFIG.api_host }, (cookies) => {
-      if (cookies === null) reject(chrome.runtime.lastError.message);
-      else resolve(cookies.filter((c) => c.name.startsWith('at-')).map((c) => c.name.substr(3)));
+      if (cookies === null) {
+        reject(chrome.runtime.lastError.message);
+      } else {
+        resolve(
+          cookies
+            // Get only Range API cookies
+            .filter((c) => c.name.startsWith('at-'))
+            // Filter cookies that have previously failed to authenticate
+            .filter((c) => !_invalidCookieCache[c.value])
+            // Filter expired cookies
+            .filter((c) => moment(c.expirationDate) - moment() / 1000 > 0)
```

Comment:
Could something expressive like https://momentjs.com/docs/#/query/is-before/ be used here? It's not obvious to me how these values cast as-is.